Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkProducer.cc

```diff
@@ -1432,7 +1430,11 @@ std::pair<double, double> AlCaHcalIsotrkProducer::storeEnergy(const HcalRespCorr
 bool AlCaHcalIsotrkProducer::notaMuon(const reco::Track* pTrack0, const edm::Handle<reco::MuonCollection>& muonh) {
   bool flag(true);
   for (reco::MuonCollection::const_iterator recMuon = muonh->begin(); recMuon != muonh->end(); ++recMuon) {
-    if (recMuon->innerTrack().isNonnull()) {
+    // If running on the AlCaHcalIsoTrk AlCaRaw data stream, then only highPurity, quality tracks are saved
+    // Thus, the reference tracker track for a muon may not exist as input here and ProductNotFound needs to be avoided
+    // If a muon's tracker track is not saved, then by definition the muon/track cannot be a candidate isolated track
+    // So catch the ProductNotFound exception and simply return true
```

Comment:
picky, but I guess now this comment line can be removed.

Reply:
no worries, and fair, it has been removed